### PR TITLE
[mongodbo] Allow any value for $unset

### PR DESCRIPTION
**Pull request by avaly**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.mongodb.com/manual/reference/operator/update/unset/
> The specified value in the $unset expression (i.e. "") does not impact the operation.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

**Review by CaselIT — APPROVED** [review]

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -1312,7 +1313,7 @@ export type UpdateQuery<TSchema> = {
     $rename?: { [key: string]: string };
     $set?: MatchKeysAndValues<TSchema>;
     $setOnInsert?: MatchKeysAndValues<TSchema>;
-    $unset?: OnlyFieldsOfType<TSchema, any, ''>;
+    $unset?: OnlyFieldsOfType<TSchema, any, any>;
```
Comment (LinusU) [inline types/mongodb/index.d.ts:1316]: Hmmm, I know that technically you can pass in anything, but I would personally prefer to be a bit more strict here.

e.g. if an instance of a class with `valueOf` or `toJSON` returning `undefined` is sent in, will the value be removed? Will `update` throw an error if a circular referencing object is passed in?

How would you feel about `'' | 1 | true` which seems to be the only usages I've seen in the wild?

That way people can't be confused by passing `false` and still having the field removed.

Comment (LinusU) [inline types/mongodb/index.d.ts:1316]: Hmmm, I know that technically you can pass in anything, but I would personally prefer to be a bit more strict here.

e.g. if an instance of a class with `valueOf` or `toJSON` returning `undefined` is sent in, will the value be removed? Will `update` throw an error if a circular referencing object is passed in?

How would you feel about `'' | 1 | true` which seems to be the only usages I've seen in the wild?

That way people can't be confused by passing `false` and still having the field removed.

**Review by LinusU — CHANGES_REQUESTED** [review]

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (avaly) [reply types/mongodb/index.d.ts:1316]: How about `string | 1 | true`?

Comment (LinusU) [reply types/mongodb/index.d.ts]: I would personally prefer `''` over `string`, I don't really see what other strings people would pass in? 🤔

It's not a hill I will die on though ☺️

**Comment by LinusU** [comment]
Not sure if it's seen easily, but I added a comment here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43888#discussion_r408644102

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (avaly) [reply types/mongodb/index.d.ts]: OK, I'll make it `''` :)

**Review by LinusU — APPROVED** [review]
Neat 👍

**Pull request merged**